pkg/provider/azure_loadbalancer_backendpool.go

```diff
@@ -105,27 +126,45 @@ func (bc *backendPoolTypeNodeIPConfig) CleanupVMSetFromBackendPoolByCondition(sl
 			}
 
 			newBackendPools[j] = bp
-			break
+			count++
+			if count == ipFamiliesCount {
+				break
+			}
 		}
 	}
 
 	for vmSetName := range vmSetNameToBackendIPConfigurationsToBeDeleted {
+		shouldRefreshLB := false
 		backendIPConfigurationsToBeDeleted := vmSetNameToBackendIPConfigurationsToBeDeleted[vmSetName]
-		backendpoolToBeDeleted := &[]network.BackendAddressPool{
-			{
-				ID: pointer.String(lbBackendPoolID),
-				BackendAddressPoolPropertiesFormat: &network.BackendAddressPoolPropertiesFormat{
-					BackendIPConfigurations: &backendIPConfigurationsToBeDeleted,
+		deleteBackendPool := func(isIPv6 bool) error {
+			backendpoolToBeDeleted := &[]network.BackendAddressPool{
+				{
+					ID: pointer.String(lbBackendPoolIDs[isIPv6]),
+					BackendAddressPoolPropertiesFormat: &network.BackendAddressPoolPropertiesFormat{
+						BackendIPConfigurations: &backendIPConfigurationsToBeDeleted,
+					},
 				},
-			},
+			}
+			// decouple the backendPool from the node
+			refresh, err := bc.VMSet.EnsureBackendPoolDeleted(service, lbBackendPoolIDs[isIPv6], vmSetName, backendpoolToBeDeleted, true)
+			if refresh {
+				shouldRefreshLB = true
+			}
+			return err
 		}
-		// decouple the backendPool from the node
-		shouldRefreshLB, err := bc.VMSet.EnsureBackendPoolDeleted(service, lbBackendPoolID, vmSetName, backendpoolToBeDeleted, true)
-		if err != nil {
-			return nil, err
+		if v4Enabled {
+			if err := deleteBackendPool(false); err != nil {
```

Comment:
Can we merge these in one so that we can update the vmss within one API call?

Reply:
Good idea. Updated.